plugins/view-resources/src/components/EditDeal.svelte

```diff
@@ -0,0 +1,364 @@
+    if (editorMixin?.editor == null && clazz.extends != null) return getEditor(clazz.extends)
+    return { editor: editorMixin.editor, pinned: editorMixin?.pinned }
+  }
+
+  let mainEditor: MixinEditor | undefined
+
+  $: getEditorOrDefault(realObjectClass, _id)
+
+  async function getEditorOrDefault (_class: Ref<Class<Doc>>, _id: Ref<Doc>): Promise<void> {
+    await updateKeys()
+    mainEditor = getEditor(_class)
+  }
+
+  async function getFieldEditor (key: KeyedAttribute): Promise<AnyComponent | undefined> {
+    const attrClass = getAttributePresenterClass(hierarchy, key.attr)
+    const clazz = hierarchy.getClass(attrClass.attrClass)
+    const mix = {
+      array: view.mixin.ArrayEditor,
+      collection: view.mixin.CollectionEditor,
+      inplace: view.mixin.InlineAttributEditor,
+      attribute: view.mixin.AttributeEditor,
+      object: undefined
+    }
+    const mixinRef = mix[attrClass.category]
+    if (mixinRef) {
+      const editorMixin = hierarchy.as(clazz, mixinRef)
+      return (editorMixin as any).editor
+    } else {
+      return undefined
+    }
+  }
+
+  function getIcon (_class: Ref<Class<Obj>> | undefined): Asset | undefined {
+    if (_class === undefined) return undefined
+    let clazz = hierarchy.getClass(_class)
+    if (clazz.icon !== undefined) return clazz.icon
+    while (clazz.extends !== undefined) {
+      clazz = hierarchy.getClass(clazz.extends)
+      if (clazz.icon != null) {
+        return clazz.icon
+      }
+    }
+    // throw new Error(`Icon not found for ${_class}`)
+  }
+
+  $: icon = getIcon(realObjectClass)
+
+  const title: string = ''
+  let rawTitle: string = ''
+
+  $: if (object !== undefined) {
+    getTitle(object).then((t) => {
+      rawTitle = t
+    })
+  }
+
+  async function getTitle (object: Doc): Promise<string> {
+    const name = (object as any).name
+    if (name !== undefined) {
+      if (hierarchy.isDerived(object._class, contact.class.Person)) {
+        return getName(client.getHierarchy(), object as Contact)
+      }
+      return name
+    }
+    const label = hierarchy.getClass(object._class).label
+    return await translate(label, {}, $themeStore.language)
+  }
+
+  const _update = (result: any): void => {
+    dispatch('update', result)
+  }
+  let panelWidth: number = 0
+  let innerWidth: number = 0
+
+  function showMenu (ev?: Event): void {
+    if (object !== undefined) {
+      showPopup(ContextMenu, { object, excludedActions: [view.action.Open] }, (ev as MouseEvent).target as HTMLElement)
+    }
+  }
+
+  $: finalTitle = title ?? rawTitle
+
+  const handleViewModeChanged = async (newMode: string) => {
+    if (newMode === undefined || (newMode === mode && currentResource !== undefined)) return
+    mode = newMode
+    currentResource = await getResource(presenterList[newMode].presenter)
+  }
+
+  const modeList: TabItem[] = [
+    { id: 'mail', labelIntl: plugin.string.Mail, action: () => handleViewModeChanged('mail') },
```

Comment:
All of this should be configurable via model extensions and provided from different plugins, like Email from gmail, Telegram from Telegram, Activity from generic one.